globals.go

```diff
@@ -80,10 +82,12 @@ type (
 
 	// Stmt is Oracle statement
 	Stmt struct {
-		conn   *Conn
-		stmt   *C.OCIStmt
-		closed bool
-		ctx    context.Context
+		conn        *Conn
+		stmt        *C.OCIStmt
+		closed      bool
+		ctx         context.Context
+		cacheKey    *C.OraText
```

Comment:
I originally had it that way, but switched because it avoided allocated on the `prepare` and on the `release`. So now we just allocate on the `prepare`, and free on the `release`. Not a big deal to switch if that is what you would prefer.

Reply:
Yes there is the extra allocation but less likely a memory leak. I think the extra allocation is worth it. Should be minimal.

Reply:
Oh, there is already a method named `query`, so we can't name it that. I'll keep the field name, but change the type.
```
func (stmt *Stmt) query(binds []bindStruct) (driver.Rows, error)